refactor: add additional comments and minor refactor to gc template (#13696)
OfficeDev/TeamsFx: templates/ts/graph-connector/README.md

# Copilot Connector Template

## Summary

> This template project uses Microsoft 365 Agents Toolkit for Visual Studio Code to simplify the process of creating a [Microsoft Copilot connector](https://learn.microsoft.com/graph/connecting-external-content-connectors-overview) that ingests data from the GitHub issues API to Microsoft Graph. It provides an end to end opinionated starting point of creating the connector, ingesting content and refreshing the ingested content.

Reviewer: ```suggestion
This template project uses Microsoft 365 Agents Toolkit for Visual Studio Code to simplify the process of creating a [Copilot connector](https://learn.microsoft.com/graph/connecting-external-content-connectors-overview) that ingests data from the GitHub issues API to Microsoft Graph. It provides an end to end opinionated starting point of creating the connector, ingesting content and refreshing the ingested content.
```

![External content in Microsoft 365 Copilot](./assets/copilot-results.png)

## Features

This template shows how to ingest data from a custom API into your Microsoft 365 tenant.

> It uses the GitHub API to provide a sample use case and is intended to be a starting point to create a PoC (Proof of Concept) to see the value of your data in M365 Copilot and then can be customized to your needs and LoB (Line of Business) APIs.

Reviewer: ```suggestion
It uses the GitHub API to provide a sample use case and is intended to be a starting point to help you bring your data in M365 Copilot and then customize it to your needs and LoB (Line of Business) APIs.
```

The template illustrates the following concepts:

- Simplify debugging and provisioning of resources with M365 Agents Toolkit for Visual Studio code
- Create external connection schema
- Support full ingestion of data
- Support incremental ingestion of data
- Visualize the external content in Microsoft 365 Copilot. 

> - Bonus: You can also add a new custom Copilot connector when you create a Declarative Agent (DA) in M365 Agents Toolkit.

Reviewer: Is this really a bonus? Or just another concept?

Author: I think this change came from changes from @RachitMalik12

![Creating a custom Copilot connector with a Declarative Agent (DA)](./assets/add-cc-with-da.png)

## Contributors

- [Sébastien Levert](https://github.com/sebastienlevert)
- [Luis Javier Fernández](https://github.com/luisjfdez)
- [Rachit Malik](https://github.com/RachitMalik12)

## Version History

Version|Date|Comments
-------|----|--------
1.0|December 03, 2024|Initial release
1.1|April 15, 2025|Additional comments and minor improvements
1.2|April 21, 2025|Rebrand and remove unecessary steps 

## Prerequisites

- [M365 Agents Toolkit for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=TeamsDevApp.ms-teams-vscode-extension)
- [Azure Functions Visual Studio Code extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azurefunctions)
- [Microsoft 365 Developer tenant](https://developer.microsoft.com/microsoft-365/dev-program) with [uploading custom apps enabled](https://learn.microsoft.com/microsoftteams/platform/m365-apps/prerequisites#prepare-a-developer-tenant-for-testing)

> - [Node@18](https://nodejs.org)

Reviewer: IIRC, this is not Node@20 by default in the toolkit

> - Have the ability to admin consent in Entra Admin Center.

Reviewer: Let's describe the minimal roles you need (Cloud App Developer or Global Admin)
